rasa/graph_components/validators/finetuning_validator.py

```diff
@@ -0,0 +1,341 @@
+        # that ensures domain and NLU training data are consistent, then we can
+        # drop this check.
+
+        if nlu:
+            nlu_data = importer.get_nlu_data()
+            domain = importer.get_domain()
+            self._compare_raise_update(
+                fingerprint_key=FINGERPRINT_NLU,
+                new_fingerprint=nlu_data.label_fingerprint(),
+                error_message=(
+                    "Cannot finetune after inserting new NLU labels or removing "
+                    "NLU labels. Please make sure that the NLU data that you use "
+                    "for finetuning contains at least one example for every label "
+                    "(i.e. intent, action name, ...) that was included in the NLU "
+                    "data used for training the model which we attempt to finetune "
+                    "now. Moreover, you must not add labels that were not included "
+                    "during training before. "
+                ),
+            )
+
+        # persist
+        if not finetuning:
+            self.persist()
+
+    def _compare_raise_update(
+        self, fingerprint_key: Text, new_fingerprint: Text, error_message: Text,
+    ) -> None:
+        """Compare fingerprints, raise if needed and update the fingerprints.
+
+        During finetuning, this method expects that it has seen fingerprints for
+        `fingerprint_key` before, compares the given `new_fingerprint` with that old
+        one and raises an error with `error_message` if needed.
+        When we're not in finetuning mode, this method just memorizes the given
+        `new_fingerprint`.
+
+        Args:
+           fingerprint_key: name of the fingerprint
+           new_fingerprint: a new fingerprint value
+           error_message: message of `InvalidConfigException` that will be raised if
+              a fingerprint is stored under `fingerprint_key` and differs from the
+              `new_fingerprint` - and we're in finetuning mode (according to the
+              execution context of this component)
+
+        Raises:
+           `InvalidConfigException` if and old fingerprint exists and differs from
+           the new one
+        """
+        if self._execution_context.is_finetuning:
+            old_fingerprint = self._fingerprints[fingerprint_key]
+            if old_fingerprint != new_fingerprint:
+                raise InvalidConfigException(error_message)
+        else:
+            self._fingerprints[fingerprint_key] = new_fingerprint
+
+    @staticmethod
+    def _get_fingerprint_of_domain_without_responses(domain: Domain) -> Text:
+        """Returns a fingerprint of a version of the given domain without responses.
+
+        Args:
+            domain: a domain
+        Returns:
+            fingerprint
+        """
+        # Do a copy of the domain to not change the actual domain (shallow is enough)
+        domain = copy.copy(domain)
+        # don't include the response texts in the fingerprint.
```

Comment:
I don't think we need these comments - this is obvious given the name of the method and the line of code.

Reply:
👍  they're left-overs from the old version